### PR TITLE
Lint with clippy in Travis builds

### PR DESCRIPTION
**Pull request by oliverlee**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,11 +16,13 @@ before_script:
   - if [ "$TRAVIS_RUST_VERSION" == "nightly" ]; then
         rustup component add --toolchain nightly rustfmt-preview;
```
Comment (mickvangelderen) [inline .travis.yml]: I'm not sure but isn't `rustfmt` currently preferred over the older `rustfmt-preview`.

File: .travis.yml [hunk 2]
```diff
@@ -16,11 +16,13 @@ before_script:
...
         which rustfmt || cargo install --force rustfmt-nightly;
+        rustup component add clippy;
```
Comment (mickvangelderen) [inline .travis.yml]: Does it matter that not all nightly versions support clippy? Can the build fail if thats the case (quite common: https://rust-lang.github.io/rustup-components-history/index.html)

**Review by mickvangelderen — DISMISSED** [review]

File: .travis.yml (hunk 1 above)
Comment (oliverlee) [reply .travis.yml]: I think I just copied some old travis script so I'll change that.

File: .travis.yml (hunk 2 above)
Comment (oliverlee) [reply .travis.yml]: Can we make a stable job that only runs clippy? Or will that fail due to use of nightly features?

Comment (oliverlee) [reply .travis.yml]: https://github.com/rust-lang/rust-clippy/#travis-ci

File: .travis.yml (hunk 1 above)
Comment (oliverlee) [reply .travis.yml]: Apparently if you look at https://static.rust-lang.org/dist/channel-rust-nightly.toml, it renames rustfmt to rustfmt-preview.

```
oliver@canopus:~/repos/dominion$  curl -s https://static.rust-lang.org/dist/channel-rust-nightly.toml | grep -A1 rename | grep rustfmt
[renames.rustfmt]
to = "rustfmt-preview"
```

**Pull request merged**